internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -86,14 +87,17 @@ func (t *Translator) ProcessBackendTrafficPolicies(backendTrafficPolicies []*egv
 			res = append(res, policy)
 
 			// Negative statuses have already been assigned so its safe to skip
-			route := resolveBTPolicyRouteTargetRef(policy, routeMap)
-			if route == nil {
+			route, resolveErr := resolveBTPolicyRouteTargetRef(policy, routeMap)
+			if route == nil && resolveErr == nil {
 				continue
 			}
 
 			// Find the Gateway that the route belongs to and add it to the
-			// gatewayRouteMap, which will be used to check policy overrides
-			for _, p := range GetParentReferences(route) {
+			// gatewayRouteMap and ancestor list, which will be used to check
+			// policy overrides and populate its ancestor status.
+			parentRefs := GetParentReferences(route)
```

Comment:
what happens if route is nil ?

Reply:
as you can see the changes in L297 and L308, the route got returned even it has resolve err in order to compute its `ancestorRefs`. so the route here will not be nil.